language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/createvar/CreateVariableCodeAction.java

```diff
@@ -129,6 +131,33 @@ CreateVariableOut getCreateVariableTextEdits(Range range, DiagBasedPositionDetai
         return new CreateVariableOut(name, types, edits, importEdits);
     }
 
+    /**
+     * Given the position details, this method will determine the expected type symbol for the required variable
+     * assignment from diagnostic properties.
+     *
+     * @param positionDetails Position details
+     * @return Optional expected type symbol
+     */
+    protected Optional<TypeSymbol> getExpectedTypeSymbol(DiagBasedPositionDetails positionDetails) {
```

Comment:
Alternatively we can extract this function to a `Function<>` and call positionDetails.diagnosticProperty(function). Refer https://github.com/ballerina-platform/ballerina-lang/blob/0e66fbf41e11c71f0d2b0211697f3d43867eb506/language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/changetype/ChangeVariableTypeCodeAction.java#L82

Reply:
Done!